pkg/cache/cache.go

```diff
@@ -57,6 +59,14 @@ type Watch struct {
 	// close the corresponding stream.
 	Value chan Response
 
+	// Type is the response type.
+	Type ResponseType
+
+	// Names for monitored resources. LDS/CDS expect empty list since they are
+	// discovered globally. RDS/EDS expect explicitly enumerated resource names
+	// as dictated by LDS/CDS.
```

Comment:
Don’t understand. More explanation..

Reply:
See the discussion on slack...

Reply:
Could you summarize it and dump it here ?

Reply:
It's pretty much what comment says. LDS/CDS request all resources. EDS/RDS request resources by name. We only reply to requested names since the rest gets ignored.

Reply:
And put this explanation above (which is clearer) in the comment :)

Reply:
Done